src/bp/core/routers/admin/workspaces.ts

```diff
@@ -78,9 +78,13 @@ export class WorkspacesRouter extends CustomRouter {
       '/:workspaceId/pipeline',
       this.asyncMiddleware(async (req, res) => {
         const { workspaceId } = req.params
-        const { pipelineId, resetStage } = req.body
+        const { pipelineId, resetStage, updateCustom } = req.body
 
-        if (pipelineId !== 'custom') {
+        if (updateCustom) {
+          const { pipeline } = req.body
```

Comment:
You can move  `pipeline` to the previous extract of `req.body`. Maybe also check if it's not undefined before merging it to the config